UnityProject/Assets/Scripts/Core/Chat/ChatRelay.cs

```diff
@@ -147,39 +147,44 @@ bool DistanceCheck(Vector3 playerPos)
 				}
 			}
 
-			//Get NPCs in vicinity
-			var npcs = Physics2D.OverlapCircleAll(chatEvent.originator.AssumedWorldPosServer(), 14f, npcMask);
-			foreach (Collider2D coll in npcs)
+
+			if (chatEvent.originator != null)
 			{
-				var npcPosition = coll.gameObject.AssumedWorldPosServer();
-				if (MatrixManager.Linecast(chatEvent.position,LayerTypeSelection.Walls,
-					 layerMask,npcPosition).ItHit ==false)
+				//Get NPCs in vicinity
+				var npcs = Physics2D.OverlapCircleAll(chatEvent.position, 14f, npcMask);
```

Comment:
You've removed the `chatEvent.originator.AssumedWorldPosServer()` so you dont need to null check anymore

Reply:
the problem is poly was picking up action text

Reply:
so I don't know if it's expecting, all the chat events to be players

Reply:
didn't someone separate messages into channels so we don't mix up action and voice?

Reply:
Polly still picks it up if I pass it

Reply:
ok, minor issue then considering this might fix the server freezing